PWGCF/MultiparticleCorrelations/Tasks/ThreeParticleCorrelations.cxx

```diff
@@ -0,0 +1,316 @@
+              }
+            } else {
+              if (A_PID[0] == 0) { // Pions
+                SECorrRegistry.fill(HIST("hSameLambdaPion_SB"), DeltaPhi, DeltaEta, collision.centFT0C(), collision.posZ(), T_Sign, associate.sign());
+              } else if (A_PID[0] == 1) { // Kaons
+                SECorrRegistry.fill(HIST("hSameLambdaKaon_SB"), DeltaPhi, DeltaEta, collision.centFT0C(), collision.posZ(), T_Sign, associate.sign());
+              } else if (A_PID[0] == 2) { // Protons
+                SECorrRegistry.fill(HIST("hSameLambdaProton_SB"), DeltaPhi, DeltaEta, collision.centFT0C(), collision.posZ(), T_Sign, associate.sign());
+              }
+            }
+          }
+        }
+      }
+    }
+    // End of the V0-Track Correlations
+  }
+  PROCESS_SWITCH(ThreePartCorr, processSame, "Process same-event correlations", true);
+
+  void processMixed(MyFilteredCollisions const& collisions, MyFilteredV0s const& v0s, MyFilteredTracks const& tracks)
+  {
+
+    // Start of the Mixed-events Correlations
+    for (const auto& [coll_1, v0_1, coll_2, track_2] : pair) {
+      for (const auto& [trigger, associate] : soa::combinations(soa::CombinationsFullIndexPolicy(v0_1, track_2))) {
+        if (V0Filters(trigger) && TrackFilters(trigger, associate)) {
+
+          T_Sign = V0Sign(trigger);
+          if (T_Sign == 1) {
+            CandMass = trigger.mLambda();
+          } else if (T_Sign == -1) {
+            CandMass = trigger.mAntiLambda();
+          }
+
+          A_PID = TrackPID(associate);
+          DeltaPhi = DeltaPhiShift(trigger.phi(), associate.phi());
+          DeltaEta = trigger.eta() - associate.eta();
+
+          if (CandMass >= 1.10 && CandMass <= 1.13) {
+            if (A_PID[0] == 0) { // Pions
+              MECorrRegistry.fill(HIST("hMixLambdaPion_SGNL"), DeltaPhi, DeltaEta, coll_1.centFT0C(), coll_1.posZ(), T_Sign, associate.sign());
+            } else if (A_PID[0] == 1) { // Kaons
+              MECorrRegistry.fill(HIST("hMixLambdaKaon_SGNL"), DeltaPhi, DeltaEta, coll_1.centFT0C(), coll_1.posZ(), T_Sign, associate.sign());
+            } else if (A_PID[0] == 2) { // Protons
+              MECorrRegistry.fill(HIST("hMixLambdaProton_SGNL"), DeltaPhi, DeltaEta, coll_1.centFT0C(), coll_1.posZ(), T_Sign, associate.sign());
+            }
+          } else {
+            if (A_PID[0] == 0) { // Pions
+              MECorrRegistry.fill(HIST("hMixLambdaPion_SB"), DeltaPhi, DeltaEta, coll_1.centFT0C(), coll_1.posZ(), T_Sign, associate.sign());
+            } else if (A_PID[0] == 1) { // Kaons
+              MECorrRegistry.fill(HIST("hMixLambdaKaon_SB"), DeltaPhi, DeltaEta, coll_1.centFT0C(), coll_1.posZ(), T_Sign, associate.sign());
+            } else if (A_PID[0] == 2) { // Protons
+              MECorrRegistry.fill(HIST("hMixLambdaProton_SB"), DeltaPhi, DeltaEta, coll_1.centFT0C(), coll_1.posZ(), T_Sign, associate.sign());
+            }
+          }
+        }
+      }
```

Comment:
In here the code is basically a replica of the same event processing. Replicating code is error prone and difficult to maintain and evolve. I would suggest avoiding replication by using the template mechanism as much as possible